include/swift/SIL/SILBitfield.h

```diff
@@ -161,7 +164,45 @@ class BasicBlockSet {
   /// Returns true if \p block was not contained in the set before inserting.
   bool insert(SILBasicBlock *block) { return !flag.testAndSet(block); }
 
-  void remove(SILBasicBlock *block) { flag.reset(block); }
+  void erase(SILBasicBlock *block) { flag.reset(block); }
+};
+
+/// An implementation of `llvm::SetVector<SILBasicBlock *,
+///                                       SmallVector<SILBasicBlock *, N>,
+///                                       BasicBlockSet>`.
+///
+/// Unfortunately it's not possible to use `llvm::SetVector` directly because
+/// the BasicBlockSet constructor needs a `SILFunction` argument.
+///
+/// Note: This class does not provide a `remove` method intentinally, because
+/// it would have a O(n) complexity.
+template <unsigned N> class BasicBlockSetVector {
```

Comment:
Can you change the name of this to reflect that one can only add and can never remove? The name suggests it is a general utility.

Reply:
Hm, I like this name. It describes best what this class is.
I even think that there shouldn't be a `remove` method in llvm:SetVector. It is O(n), but people use it as if it was O(0).